### ISSUE TITLE
Add missing supports for kubernetes client stable and gold

### ISSUE THREAD

**Issue by mbohlool** [kubernetes-incubator/client-python#166]
according to new [kubernetes client guidelines](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/csi-new-client-library-procedure.md#client-capabilities) python client is beta support and has silver capabilities. We are almost there to have gold capabilities and stable support. missing parts are:

Gold:
- [ ] Support attach, port-forward calls
- [ ] Proto encoding
- [ ] Generated examples

Stable:
- [ ] Support level documented per-platform
- [ ] Deprecation policy (backwards compatibility guarantees documented)
- [ ] Release process well documented
- [ ] N documented users of the library

**Comment by dims** [comment]
@mbohlool what does "proto encoding" mean?

**Comment by mbohlool** [comment]
@lavalamp should know more about this, but I assume it means we encode the request body with proto instead of json.

**Comment by lavalamp** [comment]
Correct, request & response bodies. Here's an example proto file: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/apps/v1beta1/generated.proto

You may not be able to determine the proto encoding from the OpenAPI spec--it'll be interesting to see you try to support these protos in a non-go language...

Also the levels were chosen so that Silver was a perfectly acceptable sort of client to have, so it's not bad to be Silver :)

It's just that if someone is going to write a high-bandwidth controller of some sort, proto is a lot more efficient.

**Comment by brendandburns** [comment]
@lavalamp what does "Generated Examples" mean in that doc? Does that mean that you have example code for the library? or that it is generated from something?

**Comment by pokoli** [comment]
Sorry if it's too obious, but does proto mean using this library: https://github.com/google/protobuf/tree/master/python ?

**Comment by mbohlool** [comment]
I looked briefly. The library needs protoc to be installed using other
means than pip. If prefer other python only solutions but if that is only
option, we should take it. Basically we want to be able to talk proto with
api server instead of json. That could include using generated proto
messages in k8s main repo. This needs some investigation before
implementation.

**Comment by BenElgar** [comment]
Just to clarify, does this mean port forwarding is not currently supported?

**Comment by micw523** [comment]
That is correct. Port forwarding is not yet supported on this client.

**Comment by mayu2010** [comment]
port-forward call still not supported in python-client now?

**Comment by TheOriginalAlex** [comment]
Is there a record of documented users of the library?  i.e. A markdown file or something.  I'd like to add to it if so.

**Comment by KnifeyMoloko** [comment]
Hey guys, any idea on how the attach / port-forward item is looking?
